src/shogun/classifier/Perceptron.h

```diff
@@ -71,29 +59,21 @@ class CPerceptron : public CLinearMachine
 		virtual const char* get_name() const { return "Perceptron"; }
 
 	protected:
-		/** registers and initializes parameters */
-			void init();
-
-		/** train classifier
-		 *
-		 * @param data training data (parameter can be avoided if distance or
-		 * kernel-based classifiers are used and distance/kernels are
-		 * initialized with train data)
-		 *
-		 * @return whether training was successful
-		 */
-		virtual bool train_machine(CFeatures* data=NULL);
+		virtual void init_model(CFeatures* data);
+		virtual void iteration();
+		virtual void end_training();
 
 	protected:
 		/** learning rate */
 		float64_t learn_rate;
-		/** maximum number of iterations */
-		int32_t max_iter;
 
 	private:
+		/** Store output during training */
+		SGVector<float64_t> m_output;
 		/** whether the hyperplane should be initialized in train_machine
 		 *
-		 * this allows to initialize the hyperplane externally using set_w and set_b
+		 * this allows to initialize the hyperplane to externally set
```

Comment:
this is a bit misleading: 
"Flag that determines whether hyper-plane is initialised by the algorithm, or not. The latter allows users to initialize the algorithm by manually setting weights and bias before training."